### DISCUSSION TITLE
Help Needed: Assembling Laser Module to CNC Shield & GRBL Changes

### DISCUSSION THREAD

**Discussion by aceride23** [arkypita/LaserGRBL#1143] [Q&A]
![IMG_20201018_044132](https://user-images.githubusercontent.com/41161809/96353204-8be86b80-10fc-11eb-9979-2067fc934aaa.jpg)
![IMG_20201018_044142](https://user-images.githubusercontent.com/41161809/96353205-8db22f00-10fc-11eb-8d35-336b8ef51f69.jpg)
![IMG_20201018_044146](https://user-images.githubusercontent.com/41161809/96353206-8ee35c00-10fc-11eb-8662-ed843ab19d89.jpg)
![IMG_20201018_044156](https://user-images.githubusercontent.com/41161809/96353208-90148900-10fc-11eb-825c-549ddbc6acd5.jpg)

Hi all, please help! I bought a laser cutter kit off Banggood, but the MKS board was defective so instead of buying another one I decided to opt for a CNC shield v3. However, this proved to be more challenging than I initially thought. I also could not find any information similar to my configuration online. So my questions are

1. How to I connect my 3500mW laser to the Arduino uno?
2. I have a dual Y, should I plug the motor in the Z or the additional axis slot?
3. What should I change on configuration.h?

Thank you in advance - this has been sitting stationary for months so would appreciate it if you can help!

**Comment by BeatSlayer** [comment]
1- one pin from the controller board for PWM, power directly from the PSU to the laser module.

2- just google "connecting stepper motors in series.

3- board type, drivers, pins, etc.

**Comment by StuartB4** [comment]
Like this. On mine in this picture the black and red go to a permanent supply, it can be the same supply as the one powering the shield if it's 12 volts. Or you need a seperate 12volt supply for the laser. The yellow and black got the the Z+ and Ground on the row of pins down the right hand side of the shield, all the pins on the far right edge are ground. 


![IMG_3844](https://user-images.githubusercontent.com/16980026/96353826-9c322380-10c7-11eb-9638-1e15dba28475.JPG)

**Comment by StuartB4** [comment]
Swap the bottom driver board in to the other socket. Unplug the motor lead and plug it in to the one next to it but turn it 180 degrees.
For the Y axis stepper motors one has to be connected the opposite way to the other on the shield. Because when you look at the machine working one is going clockwise and the other is counter clockwise when the shafts are facing each other.

**Comment by StuartB4** [comment]
You should'nt need to change anything in config.h. Just flash the Arduino with grbl 1.1h and then change the settings in grbl configuration on the grbl menu in LaserGRBL.

**Comment by StuartB4** [comment]
The heatsink on the bottom left motor driver board is really close to the edge, make sure it's not touching the pin heads underneath, it will short out if it is.

**Comment by aceride23** [comment]
Hi @StuartB4 and @BeatSlayer , thanks for your pointers

![IMG_20201018_213817](https://user-images.githubusercontent.com/41161809/96369234-58085700-118b-11eb-8f43-1a2e44e2ad13.jpg)


I have connected the printer in the way you described. I have also flashed it with GRBL 1.1. However, currently only the X axis can move. 

Forgot to mention that I mainly use a Mac (Openbuilds), however have taken out a Windows pc to see if it would work with LaserGrbl. On laser GRBL v3.8.0 the Arduino wouldn't even connect. Any ideas?

Thanks again for all your help. 

Edit: installed the driver and it's now connected.

**Comment by StuartB4** [comment]
Have you installed the driver on the PC. There's one on the Tools menu in LaserGRBL.

**Comment by StuartB4** [comment]
I noticed on the image you posted at the top of this post, you have the yellow and white lead connected to the PWM socket on the laser board. 
You have the Yellow on the Ground pin on the laser board, but on the shield Z+ it's the wrong way round on the image you posted today. The yellow wire should be on the ground pin on the shield to match and White on Z+.

**Comment by StuartB4** [comment]
Also the driver board you swapped to the other socket is the opposite way round to what it was before. It's upside down, that may be why the Y axis is not moving.

**Comment by StuartB4** [comment]
![IMG_38944](https://user-images.githubusercontent.com/16980026/96371815-9b4ad180-115b-11eb-92ad-e0e1bd614eca.jpg)

**Comment by aceride23** [comment]
![IMG_20201018_230448](https://user-images.githubusercontent.com/41161809/96371827-5b551000-1196-11eb-8691-e24ed1f6239b.jpg)
![IMG_20201018_230443](https://user-images.githubusercontent.com/41161809/96371834-60b25a80-1196-11eb-8517-0d38822244e9.jpg)
![IMG_20201018_230553](https://user-images.githubusercontent.com/41161809/96371876-922b2600-1196-11eb-8245-238e3d731eee.jpg)

Hi @StuartB4 I've swapped the stepper around und but still not luck. My laser doesn't seem to be lighting up either :/ Do you think it could be these jumpers?

Edit: Thanks! The laser is working now!

**Comment by aceride23** [comment]
![16030341011452131409972095680108](https://user-images.githubusercontent.com/41161809/96372072-c4895300-1197-11eb-9463-3a30c5bc2088.jpg)
 
Perfect laser working now. Thanks @StuartB4 :) just y axis left now!

**Comment by StuartB4** [comment]
The little black jumpers  under the motor driver boards are for microstepping. They break up bigger steps from the motor to smaller steps, so theres more steps for 1 revolution of the motor, you get smoother movement but less torque from the motor.
As for the Y axis, does it do anything at all, or are both motors dead.

**Comment by StuartB4** [comment]
Also, the little jumper on the EN/GND pins is not needed, thats for external driver boards. Take it off and just place it on the outer pin so you dont lose it.

**Comment by aceride23** [comment]
Both motors aren't moving, but I just checked they are both working individually. Just checking now to see whether my stepper drivers are dead.

Edit: Interestingly, when I swapped out the two stepper motors for Y and A with my other TMC2209s (which did not make a difference - y-axis still no movement), the X-axis also stopped working... Checked the wires, they're not faulty.

**Comment by StuartB4** [comment]
Thats strange. 

I don't think it would make any difference to yours, but I noticed on mine the X and the bottom Y motors plugs are connected the same way. 
The top Y has the plug rotated 180 degrees.

![image](https://user-images.githubusercontent.com/16980026/96373507-d271b080-1164-11eb-93c9-933d208e26d5.png)

**Comment by aceride23** [comment]
> Thats strange.
> 
> I don't think it would make any difference to yours, but I noticed on mine the X and the bottom Y motors plugs are connected the same way.
> The top Y has the plug rotated 180 degrees.
> 
> ![image](https://user-images.githubusercontent.com/16980026/96373507-d271b080-1164-11eb-93c9-933d208e26d5.png)

Just tried to mimic yours. No luck.. Is your board Protoneer v 3 or 3.5? Mine's v3.0

Edit: Just realised that you have a v3.51, which is probably why mine isn't working properly...

**Comment by StuartB4** [comment]
The only difference between your board and mine really is mine uses the SpnEn (Spindle Enable) pin for the PWM signal and yours uses the Z+ or Z-, thats because grbl changed the pin assignments from one version to another.

The other difference's are there is no glass fuse on mine, on your's it's between 2 of the stepper driver sockets on the left.  It's not needed apparently so they ditched it. 

There is resistor and capacitor filtering on mine to help prevent false triggering for limit switches.
Apart from that they both work the same. 

I've only just upgraded from the v3 to the 3.51, but I did'nt have to change or modify anything to use the existing setup I had. 
Just the PWM pin was different. So if your stepper drivers and motors and power supply are ok it should work how it is. 

Is your power supply strong enough to run the motors and the laser.
Also have you tried setting the current for the motors on the stepper driver boards. 
You have to do it with the motors disconnected from the shield. 

This is how to do it, but you will need a multimeter. 
https://www.youtube.com/watch?v=YJSmrbeZUp4

**Comment by aceride23** [comment]
> https://www.youtube.com/watch?v=YJSmrbeZUp4

Thanks I'll have a look after quarantine! Left my multimeter behind.. :(

**Comment by StuartB4** [comment]
Ok.
